File: Week 1 C/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt the user for the height of the pyramid and ensure it's between 1 and 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Build the pyramid
    for (int i = 0; i < height; i++)
    {
        // Print the spaces on the left side of the pyramid
        for (int j = height - i - 1; j > 0; j--)
        {
            printf(" ");
        }

        // Print the left half of the pyramid
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        // Print the gap between the two halves of the pyramid
        printf("  ");

        // Print the right half of the pyramid
        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }

        // Move to the next line for the next row of the pyramid
        printf("\n");
    }
}

File: Week 6 Python/sentimental-mario-more/mario.py
# Prompt the user to enter a number between 1 and 8
# and continue prompting until a valid number is entered
n = 0
while n < 1 or n > 8:
    try:
        n = int(input("Height: "))
    except ValueError:
        print("Not a positive integer no greater than 8")


# Loop through each row of the pyramid
for i in range(n):
    # Print the appropriate number of spaces to left-align the pyramid
    for j in range(n - i - 1):
        print(" ", end="")

    # Print the appropriate number of hashes for the current row
    for k in range(i + 1):
        print("#", end="")
    print("  ", end="")
    for l in range(i + 1):
        print("#", end="")

    # Move to the next line to start the next row
    print()